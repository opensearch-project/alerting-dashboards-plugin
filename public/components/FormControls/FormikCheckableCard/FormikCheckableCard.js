/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import { EuiCheckableCard } from '@elastic/eui';

import FormikInputWrapper from '../FormikInputWrapper';
import FormikFormRow from '../FormikFormRow';

const FormikCheckableCard = ({
  name,
  formRow = false,
  fieldProps = {},
  rowProps = {},
  inputProps = {},
}) => (
  <FormikInputWrapper
    name={name}
    fieldProps={fieldProps}
    render={({ field, form }) =>
      formRow ? (
        <FormikFormRow name={name} form={form} rowProps={rowProps}>
          <FieldCheckableCard name={name} form={form} field={field} inputProps={inputProps} />
        </FormikFormRow>
      ) : (
        <FieldCheckableCard name={name} form={form} field={field} inputProps={inputProps} />
      )
    }
  />
);

const FieldCheckableCard = ({
  name,
  form,
  field,
  inputProps: { onChange, id, label, ...rest },
}) => (
  <EuiCheckableCard
    name={name}
    id={id}
    label={label}
    {...field}
    {...rest}
    onChange={(e) =>
      typeof onChange === 'function' ? onChange(e, field, form) : field.onChange(e)
    }
  />
);

FormikCheckableCard.propTypes = {
  name: PropTypes.string.isRequired,
  formRow: PropTypes.bool,
  fieldProps: PropTypes.object,
  rowProps: PropTypes.object,
  inputProps: PropTypes.object,
};

FieldCheckableCard.propTypes = {
  name: PropTypes.string.isRequired,
  form: PropTypes.object.isRequired,
  inputProps: PropTypes.object.isRequired,
  field: PropTypes.object.isRequired,
};

export default FormikCheckableCard;
