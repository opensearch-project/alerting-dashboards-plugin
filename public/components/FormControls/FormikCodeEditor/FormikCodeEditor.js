/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { EuiCodeEditor } from '@elastic/eui';

import FormikInputWrapper from '../FormikInputWrapper';
import FormikFormRow from '../FormikFormRow';

const FormikCodeEditor = ({
  name,
  formRow = false,
  fieldProps = {},
  rowProps = {},
  inputProps = {},
}) => (
  <FormikInputWrapper
    name={name}
    fieldProps={fieldProps}
    render={({ field, form }) =>
      formRow ? (
        <FormikFormRow name={name} form={form} rowProps={rowProps}>
          <CodeEditor name={name} form={form} field={field} inputProps={inputProps} />
        </FormikFormRow>
      ) : (
        <CodeEditor name={name} form={form} field={field} inputProps={inputProps} />
      )
    }
  />
);

const CodeEditor = ({ name, form, field, inputProps: { onBlur, onChange, ...rest } }) => (
  <EuiCodeEditor
    id={name}
    onChange={(string) => {
      onChange(string, field, form);
    }}
    onBlur={(e) => {
      onBlur(e, field, form);
    }}
    value={field.value}
    {...rest}
  />
);

FormikCodeEditor.propTypes = {
  name: PropTypes.string.isRequired,
  formRow: PropTypes.bool,
  fieldProps: PropTypes.object,
  rowProps: PropTypes.object,
  inputProps: PropTypes.object,
};

CodeEditor.propTypes = {
  name: PropTypes.string.isRequired,
  inputProps: PropTypes.object.isRequired,
  form: PropTypes.object.isRequired,
  field: PropTypes.object.isRequired,
};

export default FormikCodeEditor;
