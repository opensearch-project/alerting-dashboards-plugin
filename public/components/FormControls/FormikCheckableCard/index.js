/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import FormikCheckableCard from './FormikCheckableCard';

export default FormikCheckableCard;
