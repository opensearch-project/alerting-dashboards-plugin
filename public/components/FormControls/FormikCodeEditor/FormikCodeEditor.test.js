/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from 'enzyme';
import { Formik } from 'formik';

import FormikCodeEditor from './FormikCodeEditor';
import FormikCheckableCard from '../FormikCheckableCard';

// FIXME: This has an issue where EuiCodeEditor is generating a random HTML id and failing snapshot test

describe('FormikCodeEditor', () => {
  test('render formRow', () => {
    const component = (
      <Formik>
        <FormikCodeEditor name="testing" formRow />
      </Formik>
    );

    expect(render(component)).toMatchSnapshot();
  });
  test('renders', () => {
    const component = (
      <Formik>
        <FormikCodeEditor name="testing" />
      </Formik>
    );

    expect(render(component)).toMatchSnapshot();
  });
});
