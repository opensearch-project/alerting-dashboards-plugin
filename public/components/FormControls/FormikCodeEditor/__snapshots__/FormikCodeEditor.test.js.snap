// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormikCodeEditor render formRow 1`] = `
<div
  class="euiFormRow euiFormRow--compressed"
  id="testing-form-row-row"
>
  <div
    class="euiFormRow__fieldWrapper"
  >
    <div
      class="euiCodeEditorWrapper"
      data-test-subj="codeEditorContainer"
    >
      <button
        class="euiCodeEditorKeyboardHint"
        data-test-subj="codeEditorHint"
        id="generated-id"
      >
        <p
          class="euiText"
        >
          Press Enter to start editing.
        </p>
        <p
          class="euiText"
        >
          When you're done, press Escape to stop editing.
        </p>
      </button>
      <div
        id="generated-id"
        style="width:500px;height:500px"
      />
    </div>
  </div>
</div>
`;

exports[`FormikCodeEditor renders 1`] = `
<div
  class="euiCodeEditorWrapper"
  data-test-subj="codeEditorContainer"
>
  <button
    class="euiCodeEditorKeyboardHint"
    data-test-subj="codeEditorHint"
    id="generated-id"
  >
    <p
      class="euiText"
    >
      Press Enter to start editing.
    </p>
    <p
      class="euiText"
    >
      When you're done, press Escape to stop editing.
    </p>
  </button>
  <div
    id="generated-id"
    style="width:500px;height:500px"
  />
</div>
`;
